syntax = "proto3";

package streaming;

message VideoFrame {
  bytes data = 1;
  int32 width = 2;
  int32 height = 3;
  int64 timestamp = 4;
}

message AudioChunk {
  bytes data = 1; 
  int32 sampleRate = 2;
  int32 bitDepth = 3;
  int32 channels = 4;
  int64 timestamp = 5;
}

message Event {
  string type = 1; 
  string content = 2;
}

message StreamData {
  oneof payload {
    VideoFrame video = 1;
    AudioChunk audio = 2;
    Event event = 3;
  }
}
